pkg/cloud/services/loadbalancer/loadbalancer.go

```diff
@@ -356,6 +363,12 @@ func (s *Service) getOrCreateMonitor(openStackCluster *infrav1.OpenStackCluster,
 		Timeout:        5,
 	}
 	monitor, err = s.loadbalancerClient.CreateMonitor(monitorCreateOpts)
+	// Skip creating monitor if it is not supported by Octavia provider
+	if strings.Contains(err.Error(), "501") {
```

Comment:
I think a proper check would be something like here [1]. I see there's no `ErrDefault501`, but you should be able to convert to `ErrUnexpectedResponseCode` and then do the check?

[1] https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/8e4cf26f716ed50abd9d3efc74d0297dfba633ae/pkg/utils/errors/errors.go#L58-L72

Reply:
Added the conversion, check if the status code is `501` record the warning and event.